src/Sarif/IResultLogWriter.cs

```diff
@@ -38,12 +38,23 @@ public interface IResultLogWriter
         /// all results have been generated.
         /// </summary>
         /// <param name="fileDictionary">
-        /// A dictionary whose keys are the URIs of scanned files and whose values provide
-        /// information about those files.
+        /// A dictionary whose keys are the strings representing the locations of scanned files
+        /// and whose values provide information about those files.
         /// </param>
         void WriteFiles(IDictionary<string, IList<FileData>> fileDictionary);
 
         /// <summary>
+        /// Write information about the logical locations where results were produced to
+        /// the log. This information may appear after the results, as the full list of
+        /// logical locations will not be known until all results have been generated.
+        /// </summary>
+        /// <param name="logicalLocationDictionary">
+        /// A dictionary whose keys are strings specifying a logical location and
+        /// whose values provide information about each component of the logical location.
+        /// </param>
+        void WriteLogicalLocations(IDictionary<string, LogicalLocationComponent[]> logicalLocationDictionary);
```

Comment:
I will certainly make it an `IList` for consistency with `WriteFiles` above.

But do note that this isn't part of the SARIF object model's API surface. It just says to converter authors "please give me an array so I can write it out". You see that its implementation doesn't actually touch the object model; it just writes out raw JSON.